### PR TITLE
SYMPHONY and Cbc

### PR DESCRIPTION
**Pull request by tkralphs**
I did a formula for SYMPHONY, too, and the issues are similar to Cbc, so I thought I might as well just create a pull request against both.

### CODE REVIEW

File: cbc.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Cbc < Formula
+  desc "Linear programming solver"
```
Comment (dpo) [inline cbc.rb]: This is still the old `desc`. I suggest we keep it to one formula per pull request when possible.

File: symphony.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  depends_on "homebrew/science/openblas" => :optional
+
+  asl_dep = (build.with? "asl") ? ["with-asl"] : []
+  glpk_dep = (build.with? "glpk") ? ["with-glpk"] : []
+  openblas_dep = (build.with? "openblas") ? ["with-openblas"] : []
+
+  depends_on "homebrew/science/mumps" => [:optional, "without-mpi"] + openblas_dep
+  depends_on "homebrew/science/suite-sparse" => [:optional] + openblas_dep
+
+  mumps_dep = (build.with? "mumps") ? ["with-openblas"] : []
```
Comment (dpo) [inline symphony.rb]: `"with-mumps"`

File: symphony.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  suite_sparse_dep = (build.with? "suite-sparse") ? ["with-openblas"] : []
```
Comment (dpo) [inline symphony.rb]: `"with-suite-sparse"`

**Comment by dpo** [comment]
Did you have a look at the `symphony.rb` in Homebrew/Science (which I'll have to remove after we merge this)? It installs some data sets and has a `test` method.

**Comment by tkralphs** [comment]
OK, I took a look at ```symphony.rb``` in Homebrew/Science and there were indeed some options that I had forgotten about. I added those and the testing stuff to the updated formula here. I'm sorry about having different version of ```cbc``` in each pull request. I screwed up by committing my the ```cbc``` formula to my ```master``` branch and then when I tried to create a new branch for ```symphony```, ```cbc``` got carried along. I'm still fumbling around a bit with git. Now I think I've merged all the changes from my master branch to this pull request so I'll close the other one. I agree that in the the future, we should have just one formula per pull request.

Some additional comments. 

* OpenMP is now default for SYMPHONY, so I guessed at how to add a "negative" option --without-openmp to *disable* it. Can you check that?
* I had to add "-lreadline" to what was in the previous formula, since the build broke without it. This is strange, since it was working before. I do recall some kind of problem involving a built-in OS X ```readline``` library. Perhaps there was one before and it has now gone away in the upgraded OS X? In general, the presence of ```readline``` should be detected automatically, as long as the headers and libraries are in a standard location. Hence, there is no option to enable/disable readline support. It is always linked if ```readline``` is present (and ```--enable-gnu-packages``` is chosen). I'm not sure why any of what's there now should be necessary (it's not on Linux), but it's working so I guess we can leave well enough alone.
* This brought to my mind that both ```clp``` and ```cbc``` can also use readline support (if ```--enable-gnu-packages``` is chosen). I can add those as a separate pull request if you agree.
* In general, I always build with ```--enable-gnu-packages``` and you had it in the old ```symphony``` formula. Should we add it to all the other formulae, too? I can maybe do that also as a single monolithic pull request if that makes sense.
* ```cbc``` can be built in parallel, like ```symphony```, but this is not the default. In my experience, it's a bit flaky in parallel and I've never tried it on OS X. Should we add that as an option?

**Comment by tkralphs** [comment]
Where do we stand on this? I just wanted to make sure we're not each waiting for the other. I was thinking we should merge this pull request and then go back and fix any lingering issues with previous formulas before moving on to the next ones.

**Comment by jonahgeorge** [comment]
Not sure the status of this pull request; however, I was able to apply it as a patch and install/use cbc without any issues. Thank you

**Pull request closed without merging**